### PR TITLE
feat: Update DatePicker disabledDays type to include Modifier[]

### PR DESCRIPTION
**Pull request by ksk5280**
Updating `DatePicker` `disabledDays` prop to accept an array of Modifiers to fix TS error in this PR: https://github.com/homebound-team/internal-frontend/pull/1794/files#diff-b5da248aa7ad502a7ca5b33777e5a23e60a778ec6084a912bf80471f381480e1R61

### CODE REVIEW

File: src/components/internal/DatePicker.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ import "./DatePicker.css";
 export interface DatePickerProps {
   value?: Date;
   onSelect: (value: Date) => void;
-  disabledDays?: Modifier;
+  disabledDays?: Modifier | Modifier[];
```
Comment (bdow) [inline src/components/internal/DatePicker.tsx:11]: **thought:** Could we make it only accept an array? We'd need to update existing implementations of this, but I think SchedulesV2's `DateCellField` is the only user of this at the moment.

Comment (ksk5280) [reply src/components/internal/DatePicker.tsx]: 👍 Updated!

File: src/components/internal/DatePicker.tsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ import "./DatePicker.css";
 export interface DatePickerProps {
   value?: Date;
   onSelect: (value: Date) => void;
-  disabledDays?: Modifier;
+  disabledDays?: Modifier[];
```
Comment (blambillotte) [inline src/components/internal/DatePicker.tsx:11]: What do you think about `Modifier | Modifier[]` to allow for either?

File: src/inputs/DateField.mock.test.tsx [hunk 3]
```diff
@@ -35,10 +35,13 @@ describe("MockDateField", () => {
   });
 
   it("forwards disabledDays prop as stringified object", async () => {
-    const testDate = new Date(2020, 0, 1)
-    const r = await render(<MockDateField label="Start Date" value={testDate} disabledDays={{ before: testDate }} onChange={() => {}} />);
+    const testDate = new Date(2020, 0, 1);
+    const disabledDays = [{ before: testDate }, () => false];
```
Comment (blambillotte) [inline src/inputs/DateField.mock.test.tsx:39]: I'm not sure I understand this noop `() => false` as a `Modifier`? it that some special case?

File: src/components/internal/DatePicker.tsx (hunk 2 above)
Comment (ksk5280) [reply src/components/internal/DatePicker.tsx:11]: I had that initially, but Brandon suggested updating to only accept an array.

File: src/inputs/DateField.mock.test.tsx (hunk 3 above)
Comment (ksk5280) [reply src/inputs/DateField.mock.test.tsx:39]: It's to represent a FunctionModifier, which I'm using to [disable days for the scheduleSettings](https://github.com/homebound-team/internal-frontend/pull/1794/files#diff-b5da248aa7ad502a7ca5b33777e5a23e60a778ec6084a912bf80471f381480e1R61).

File: src/components/internal/DatePicker.tsx (hunk 2 above)
Comment (bdow) [reply src/components/internal/DatePicker.tsx:11]: Haha, I had just asked to see if we can only do the array type. I'm not sold on one way or the other, so happy to move back to what you originally had @ksk5280.

**Review by bdow — APPROVED** [review]

File: src/components/internal/DatePicker.tsx (hunk 2 above)
Comment (blambillotte) [reply src/components/internal/DatePicker.tsx]: Ha I don't have a strong opinion - my original intention was to just forward this prop and not deviate from `react-day-picker`'s implementation so the docs would still make sense with our wrapper... so this was a miss from me that I made it more restrictive

Comment (ksk5280) [reply src/components/internal/DatePicker.tsx]: I think it seems good to keep it flexible, so I reverted back to `Modifier | Modifier[]`

**Review by bdow — APPROVED** [review]
Looks great. Thanks for adding the stories

**Pull request merged**